logstash-core/lib/logstash/webserver.rb

```diff
@@ -110,3 +122,11 @@ def start_webserver(port)
     end
   end
 end
+
+# Reopen the puma class to create a scoped STDERR and STDOUT
+# This operation is thread safe since its done at the class level
+# and force JRUBY to flush his classes cache.
+module Puma
```

Comment:
can we perhaps do this in https://github.com/elastic/logstash/tree/master/logstash-core/lib/logstash/patches ?